docs/source/conf.py

```diff
@@ -166,7 +166,18 @@
     ),
 ]
 
-intersphinx_mapping = {"python": ("https://docs.python.org/3", None)}
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3", None),
+    "distributed": ("https://distributed.dask.org/en/stable", None),
+    "lightgbm": ("https://lightgbm.readthedocs.io/en/latest", None),
+    "matplotlib": ("https://matplotlib.org/stable", None),
+    "numpy": ("https://numpy.org/doc/stable", None),
+    "scipy": ("https://docs.scipy.org/doc/scipy", None),
+    "sklearn": ("https://scikit-learn.org/stable", None),
+    "torch": ("https://pytorch.org/docs/stable", None),
```

Comment:
[Notes] `torch.distributed.ProcessGroup` was not linked to the pytorch docs. The URL seems to be correct, and we may need to update the docstring/type hints in a follow-up PR.
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/3255979/216489259-b369c94e-9b9c-4d98-8644-73b6b6ccaa6b.png">